Scylla Manager crashed and cannot start (could be due to https://github.com/scylladb/scylla-manager/pull/3941 )

We run Scylla Manager 3.3.0 with Scylla 5.4.9. Today I noticed that the manager is crashed and is in CrashLoopBackoff after 24 failed restarts.

The logs:
```
{"L":"INFO","T":"2024-08-15T08:56:33.135Z","N":"backup.index","M":"Done indexing snapshot files","duration":"8.050019781s","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.241.174","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.243.109","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.241.174","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.243.109","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.241.175","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.241.175","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.249.238","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.249.238","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Removing duplicated files from local snapshot","host":"10.7.241.130","_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:33.138Z","N":"backup.deduplicate","M":"Setting rate limit","host":"10.7.241.130","limit":100,"_trace_id":"28VLtEwgRxmNtUplRi_qrQ"}
{"L":"INFO","T":"2024-08-15T08:56:38.678Z","N":"http","M":"GET /api/v1/clusters","from":"10.6.249.1:39518","status":200,"bytes":700,"duration":"7ms","_trace_id":"9O5dDq_dQAO1BsYAZhDeMQ"}
{"L":"INFO","T":"2024-08-15T08:56:48.677Z","N":"http","M":"GET /api/v1/clusters","from":"10.6.249.1:54724","status":200,"bytes":700,"duration":"6ms","_trace_id":"sx_qPHGWTmy3YGdDCe7dRw"}
{"L":"INFO","T":"2024-08-15T08:56:58.680Z","N":"http","M":"GET /api/v1/clusters","from":"10.6.249.1:36986","status":200,"bytes":700,"duration":"9ms","_trace_id":"J5MXev4KQHGmBMguVD7RGg"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1174b4d]

goroutine 2125 [running]:
github.com/scylladb/scylla-manager/v3/pkg/scyllaclient.(*Client).RcloneDeletePaths(0xc000002000, {0x2d81720, 0xc000371500}, {0xc000ba3130, 0xc}, {0xc000
b2b8c0?, 0xc01515a000?}, {0xc01515d008, 0x2e2, 0x2ff})
    github.com/scylladb/scylla-manager/v3/pkg/scyllaclient/client_rclone.go:384 +0x20d
github.com/scylladb/scylla-manager/v3/pkg/scyllaclient.(*Client).RcloneDeletePathsInBatches(0xc000002000, {0x2d81720, 0xc000371500}, {0xc000ba3130, 0xc}
, {0xc000b2b8c0, 0x5a}, {0xc01515a008, 0x2e2, 0x2e2}, ...)
    github.com/scylladb/scylla-manager/v3/pkg/scyllaclient/client_rclone.go:353 +0x1e7
github.com/scylladb/scylla-manager/v3/pkg/service/backup.(*worker).deduplicateHost.func2(0x0?)
    github.com/scylladb/scylla-manager/v3/pkg/service/backup/worker_deduplicate.go:101 +0x8af
github.com/scylladb/scylla-manager/v3/pkg/util/parallel.Run.func1()
    github.com/scylladb/scylla-manager/v3/pkg/util/parallel/parallel.go:72 +0xb2
created by github.com/scylladb/scylla-manager/v3/pkg/util/parallel.Run in goroutine 1932
    github.com/scylladb/scylla-manager/v3/pkg/util/parallel/parallel.go:57 +0xe8
Stream closed EOF for scylla-manager/scylla-manager-984b44d9c-4m8km (scylla-manager)
```

The cluster repair task was in progress.

--- a/pkg/scyllaclient/client_rclone.go
+++ b/pkg/scyllaclient/client_rclone.go
@@ -381,7 +381,10 @@ func (c *Client) RcloneDeletePaths(ctx context.Context, host, remoteDir string, 
 		},
 	}
 	res, err := c.agentOps.OperationsDeletepaths(&p)
-	return res.Payload.Deletes, err
+	if err != nil {
+		return 0, err
+	}
+	return res.Payload.Deletes, nil
 }
 
 // RcloneDiskUsage get disk space usage.

